core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -17,27 +17,178 @@
 
 package org.apache.spark.scheduler
 
+import java.io._
 import java.nio.ByteBuffer
+import java.util.Properties
 
-import org.apache.spark.util.SerializableBuffer
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+import org.apache.spark.serializer.SerializerInstance
+import org.apache.spark.util.{ByteBufferInputStream, ByteBufferOutputStream}
 
 /**
  * Description of a task that gets passed onto executors to be executed, usually created by
  * [[TaskSetManager.resourceOffer]].
  */
-private[spark] class TaskDescription(
-    val taskId: Long,
-    val attemptNumber: Int,
-    val executorId: String,
-    val name: String,
-    val index: Int,    // Index within this task's TaskSet
-    _serializedTask: ByteBuffer)
-  extends Serializable {
+private[spark] class TaskDescription private(
+  val taskId: Long,
+  val attemptNumber: Int,
+  val executorId: String,
+  val name: String,
+  val index: Int,
+  val taskFiles: mutable.Map[String, Long],
+  val taskJars: mutable.Map[String, Long],
+  private var task_ : Task[_],
+  private var taskBytes: InputStream,
+  private var taskProps: Properties) {
+
+  def this(taskId: Long,
+    attemptNumber: Int,
```

Comment:
nit: each parameter on its own line (even the first one), and double indent all params